src/ucp/core/ucp_worker.c

```diff
@@ -1394,8 +1422,11 @@ ucs_status_t ucp_worker_iface_open(ucp_worker_h worker, ucp_rsc_index_t tl_id,
         ucp_worker_iface_add_distance(&wiface->attr, &wiface->distance);
     }
 
-    ucp_worker_iface_get_memory_distance(wiface, &distance);
-    ucp_worker_iface_add_distance(&wiface->attr, &distance);
+    is_numa_aware = ucp_worker_iface_is_numa_aware(wiface);
+    if (is_numa_aware) {
+        ucp_worker_iface_get_memory_distance(wiface, &distance);
```

Comment:
move the logic of host mem check inside ucp_worker_iface_get_memory_distance
if not host mem it will return default (neutral) distance